io/subsystem/src/main/java/org/wildfly/extension/io/IOSubsystemTransformers.java

```diff
@@ -44,10 +47,19 @@ public String getSubsystemName() {
     public void registerTransformers(SubsystemTransformerRegistration registration) {
         ChainedTransformationDescriptionBuilder chainedBuilder = TransformationDescriptionBuilder.Factory.createChainedSubystemInstance(registration.getCurrentSubsystemVersion());
 
-        // Current 3.0.0 to 2.0.0, aka EAP 7.0.0
-        buildTransformers_2_0(chainedBuilder.createBuilder(registration.getCurrentSubsystemVersion(), VERSION_2_0));
+        // Current 3.0.0 to 2.0.0, aka EAP 7.1.0
+        buildTransformers_3_0(chainedBuilder.createBuilder(VERSION_3_0, VERSION_2_0));
+        // 2.0.0 to 1.0.0, aka EAP 7.0.0
+        buildTransformers_2_0(chainedBuilder.createBuilder(VERSION_2_0, VERSION_1_0));
 
-        chainedBuilder.buildAndRegister(registration, new ModelVersion[]{VERSION_2_0});
+        chainedBuilder.buildAndRegister(registration, new ModelVersion[]{ VERSION_3_0, VERSION_2_0 });
```

Comment:
That does not seem correct, the version chain goes from the oldest to the newest ("current version excluded").
It should be:

```
chainedBuilder.buildAndRegister(registration, new ModelVersion[]{ VERSION_2_0, VERSION_3_0 });
```

Reply:
I think the error in `IOSubsystemTransformerTestCase.testRejectingTransformersEAP_7_1_0` is caused by the wrong version chain order.

Reply:
This didn't seem to fix the problem... I thought the transformers were meant to go from newer to older though (as per the comments)?  Maybe I need to swap the order in createBuilder() as well?